activation/atxdb.go

```diff
@@ -178,23 +189,23 @@ func (db *DB) createTraversalActiveSetCounterFunc(countedAtxs map[string]types.A
 
 	traversalFunc := func(b *types.Block) (stop bool, err error) {
 
-		// don't count ATXs in blocks that are not destined to the prev epoch
-		if b.LayerIndex.GetEpoch(db.LayersPerEpoch) != epoch-1 {
+		// count unique ATXs
+		if b.ActiveSet == nil {
 			return false, nil
 		}
-
-		// count unique ATXs
-		for _, id := range b.ATXIDs {
+		for _, id := range *b.ActiveSet {
 			atx, err := db.GetAtxHeader(id)
 			if err != nil {
 				log.Panic("error fetching atx %v from database -- inconsistent state", id.ShortString()) // TODO: handle inconsistent state
 				return false, fmt.Errorf("error fetching atx %v from database -- inconsistent state", id.ShortString())
 			}
 
+			// todo: should we accept only eopch -1 atxs?
```

Comment:
you mean only atxs from the previous epoch? I think if we not limit that at all we might get atxs we don't have yet and need to sync them.